src/mbgl/renderer/layers/render_raster_layer.cpp

```diff
@@ -31,4 +36,35 @@ bool RenderRasterLayer::hasTransition() const {
     return unevaluated.hasTransition();
 }
 
+void RenderRasterLayer::uploadBuckets(gl::Context& context, RenderSource* source) {
+    if (renderTiles.size() > 0) {
```

Comment:
Reversing the conditional here would incur an unnecessary `dynamic_cast` for the most frequent use cases for a raster layer which will tend to be raster tile source. 

👍 - for base class usage.